File: Python/8py 2/maps.py
"""
|--------------------------------------------------------------------------
| Beginner - Lost Without a Map
|--------------------------------------------------------------------------
|
"""

def maps(a):
   return list(map(lambda x: x * 2, a))


maps = lambda a: [2*n for n in a]


def maps(a):
   return [2 * x for x in a]


def maps(a):
   if len(a) > 0:
      for i in range(len(a)):
         a[i] = a[i] * 2
   return a


def maps(a):
   return list(map((2).__mul__, a))


print(maps([1, 2, 3]), [2, 4, 6])
print(maps([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]), [0, 2, 4, 6, 8, 10, 12, 14, 16, 18])
print(maps([]), [])

File: PHP/8php 2/maps.php
<?php
/*
|--------------------------------------------------------------------------
| Beginner - Lost Without a Map
|--------------------------------------------------------------------------
|
*/

function maps($x)
{
   return array_map(function($i) { return $i * 2; }, $x);
}


function maps($x)
{
   $numbers = [];
   foreach ($x as $number) {
      $numbers[] = $number * 2;
   }
   return $numbers;
}


function maps($x)
{
   foreach ($x as $data) {
      $hasil[] = $data + $data;
   } return $hasil;
}


print_r(maps([1, 2, 3]));
print_r(maps([4, 1, 1, 1, 4]));
print_r(maps([2, 2, 2, 2, 2, 2]));